common/utils/tests/CMakeLists.txt

```diff
@@ -0,0 +1,20 @@
+#
+# Copyright (c) 2019, NVIDIA CORPORATION.  All rights reserved.
+#
+# NVIDIA CORPORATION and its licensors retain all intellectual property
+# and proprietary rights in and to this software, related documentation
+# and any modifications thereto.  Any use, reproduction, disclosure or
+# distribution of this software and related documentation without an express
+# license agreement from NVIDIA CORPORATION is strictly prohibited.
+#
+
+project(cgautilstests)
```

Comment:
I believe that's how CMake binaries need to be setup?
e.g.
https://github.com/clara-genomics/ClaraGenomicsAnalysis/blob/dev-v0.4.0/cudaaligner/tests/CMakeLists.txt

We also don't have any tests for our other utils. So I intend to grow this for tests around utility functions.

Reply:
You can definitely have multiple binaries in a project in CMake. But maybe it is due to our cmake structure... We should look into this at some point.

Reply:
Yep, can certainly have multiple binaries within a single project. In reality the tests folder doesn't need to be independent project since we set the `project` at the module level once. But that's somehow become the convention within our repo (for single binaries, project and target have become the same)

Reply:
Is there a reason why we do that? Otherwise I'd remove them everywhere - these are really not separate projects, but tests for a project.

Reply:
No good reason, was going to address in a separate PR. but small enough so I changed it in this one itself.